.eslintrc.js

```diff
@@ -400,10 +400,185 @@ module.exports = {
+        'no-useless-concat': 'error',
+        'no-useless-computed-key': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-useless-escape': 'warn',
+        'no-useless-rename': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-useless-return': 'warn',
+        // This will be turned on after bug fixers are mostly complete
+        // 'no-void': 'warn',
+        'one-var-declaration-per-line': 'error',
+        'prefer-object-spread': 'error',
+        'prefer-promise-reject-errors': 'error',
+        'prefer-rest-params': 'error',
+        'prefer-spread': 'error',
+        // This style will be turned on after most bugs are fixed
+        // 'prefer-template': 'warn',
+        // This style will be turned on after most bugs are fixed
+        // quotes: ['warn', 'single', { avoidEscape: true }],
+        'react/boolean-prop-naming': 'error',
+        'react/button-has-type': 'error',
+        'react/forbid-dom-props': 'error',
+        // This will go from warn to error when this is fixed:
+        // https://github.com/elastic/ingest-dev/issues/468
+        'react/no-access-state-in-setstate': 'warn',
+        // This style will be turned on after most bugs are fixed
+        // 'react/no-children-prop': 'warn',
+        'react/no-danger-with-children': 'error',
+        'react/no-deprecated': 'error',
+        'react/no-did-mount-set-state': 'error',
+        // This will go from warn to error when this is fixed:
+        // https://github.com/elastic/ingest-dev/issues/468
+        'react/no-did-update-set-state': 'warn',
+        'react/no-direct-mutation-state': 'error',
+        'react/no-find-dom-node': 'error',
+        'react/no-redundant-should-component-update': 'error',
+        'react/no-render-return-value': 'error',
+        'react/no-typos': 'error',
+        'react/no-string-refs': 'error',
+        'react/no-this-in-sfc': 'error',
+        // This can go from warn to error once this is fixed
+        // https://github.com/elastic/ingest-dev/issues/467
+        'react/no-unescaped-entities': 'warn',
+        'react/no-unsafe': 'error',
+        'react/no-unused-prop-types': 'error',
+        'react/no-unused-state': 'error',
+        // will introduced after the other warns are fixed
+        // 'react/sort-comp': 'error',
+        'react/void-dom-elements-no-children': 'error',
+        'react/jsx-boolean-value': ['error', 'warn'],
+        // will introduced after the other warns are fixed
+        // 'react/jsx-no-bind': 'error',
+        'react/jsx-no-comment-textnodes': 'error',
+        // will be introduced to fix missing i18n keys
+        // 'react/jsx-no-literals': 'warn',
+        'react/jsx-no-target-blank': 'error',
+        'react/jsx-fragments': 'error',
+        'react/jsx-sort-default-props': 'error',
+        // might be introduced after the other warns are fixed
+        // 'react/jsx-sort-props': 'error',
```

Comment:
From the existing patterns in the codebase, I think the team would be happy to have this enabled, albeit with the [callbacksLast](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md#callbackslast) and [reservedFirst](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md#reservedfirst) props enabled as well.